testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -214,6 +214,69 @@ public void testSparkParquetTimestampCompatibility(String sparkTimestampFormat,
+    public void testSparkParquetBloomFilterCompatibility()
+    {
+        String sparkTableNameWithBloomFilter = "test_spark_parquet_bloom_filter_compatibility_enabled_" + randomNameSuffix();
+        String sparkTableNameNoBloomFilter = "test_spark_parquet_bloom_filter_compatibility_disabled_" + randomNameSuffix();
+
+        // disable dictionary predicate when testing bloom filter predicate
+        onSpark().executeQuery(
+                String.format("CREATE TABLE %s (testInteger INT, testLong Long, testString STRING, testDouble DOUBLE, testFloat FLOAT, testChar CHAR(50)) ", sparkTableNameWithBloomFilter) +
+                        "USING PARQUET OPTIONS (" +
+                        "'parquet.bloom.filter.enabled'='true'," +
+                        "'parquet.enable.dictionary'='false'" +
+                        ")");
+        onSpark().executeQuery(
+                String.format("CREATE TABLE %s (testInteger INT, testLong Long, testString STRING, testDouble DOUBLE, testFloat FLOAT, testChar CHAR(50)) ", sparkTableNameNoBloomFilter) +
+                        "USING PARQUET OPTIONS (" +
+                        "'parquet.bloom.filter.enabled'='false'," +
+                        "'parquet.enable.dictionary'='false'" +
+                        ")");
+        String[] sparkTables = new String[] {sparkTableNameWithBloomFilter, sparkTableNameNoBloomFilter};
+        String[] trinoTables = new String[] {
+                format("%s.default.%s", TRINO_CATALOG, sparkTableNameWithBloomFilter),
+                format("%s.default.%s", TRINO_CATALOG, sparkTableNameNoBloomFilter)};
+
+        // control number of spark output files via hint: https://issues.apache.org/jira/browse/SPARK-24940
+        // contain values such as aaaaaaaaaaa and zzzzzzzzzzz, this made sure file level statistics: min and max won't take effect
+        for (String sparkTable : sparkTables) {
+            onSpark().executeQuery(format(
+                    "INSERT INTO %s " +
+                            "SELECT /*+ REPARTITION(1) */  testInteger, testLong, testString, testDouble, testFloat, testChar FROM VALUES " +
+                            "  (-999999, -999999, 'aaaaaaaaaaa', -9999999999.99D, -9999999.9999F, 'aaaaaaaaaaa')" +
+                            ", (3, 30, 'fdsvxxbv33cb', 97662.2D, 98862.2F, 'fdsafvvs')" +
+                            ", (5324, 2466, 'refgfdfrexx', 8796.1D, -65496.1F, 'vfdsf22')" +
+                            ", (999999, 9999999999999, 'zzzzzzzzzzz', 9999999999.99D, -9999999.9999F, 'zzzzzzzzzzz')" +
+                            ", (9444, 4132455, 'ff34322vxff', 32137758.7892D, 9978.129887F,'jkmlkkjl') AS DATA(testInteger, testLong, testString, testDouble, testFloat, testChar)",
+                    sparkTable));
+        }
+
+        // explicitly make sure using bloom filter statistics
+        onTrino().executeQuery("set session hive.parquet_use_bloom_filter=true");
+        assertTrinoBloomFilterTableSelectResult(trinoTables);
```

Comment:
can we check that the bloom filter is actually effective?

with bloom 
```
trino> set session hive.parquet_use_bloom_filter=true;
SET SESSION

trino> explain analyze select * from hive.default.inbloom where uuid='fa90997b-12bc-4395-9f72-8ec583550bbb';
                                                                             Query Plan                                                                              
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SOURCE]                                                                                                                                                 
     CPU: 94.94ms, Scheduled: 591.88ms, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 0 rows (0B); per task: avg.: 0.00 std.dev.: 0.00, Output: 0 rows (0B) 

```
No scan has been performed while using bloom when searching for a non-existent value ` Input: 0 rows`

without bloom

```
trino> set session hive.parquet_use_bloom_filter=false;
SET SESSION
trino> explain analyze select * from hive.default.inbloom where uuid='fa90997b-12bc-4395-9f72-8ec583550bbb';
                                                                                    Query Plan                                                                                    
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SOURCE]                                                                                                                                                              
     CPU: 693.38ms, Scheduled: 3.46s, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 999999 rows (39.10MB); per task: avg.: 999999.00 std.dev.: 0.00, Output: 0 rows (0B) 
```

Notice the full table scan performed while not using bloom when searching for a non-existent value
`Input rows: 999999`

Reply:
ok, let me try if product test is able to get the query statistics. If not I will do parsing.

Reply:
I would rather avoid complicating the product test with queries on system connector, a query runner test on an existing file would be simpler to understand and maintain.

Reply:
Hi, seems doTestSparkBucketedTableValidation() used readTable() call and made call to " MaterializedResult(allRows.build(), getTypes(columnHandles))". which means MaterializedResult#statementStats is set as Optional.empty()(which contains statistics such as physicalInputBytes) I will look into how to use TestingTrinoClient but takes sometime.

Meanwhile, I will try to address the rest 2 commits, if you think it is good to merge please let me know. Otherwise please wait till I figure out testing this feature using TestingTrinoClient and able to get statementStats

Reply:
Even if I were able to get MaterializedResult#statementStats, I think I will look into physicalInputBytes, processedRows properties. I am not sure whether Trino use the actual processed physically read in row or column chunk row count to update those fields. So it is possible from processedRows statistics, Trino test  can't tell the difference. The test can tell the difference from physicalInputBytes, due to the extra 1MB bloom filter reading.

Reply:
See `BaseOrcWithBloomFiltersTest` for an example of a query runner test where we validate pruning by looking at query stats. The main difference would be that we use a pre-existing file instead of writing it through trino.
You can try doing that, if turns out to be not straightforward, we can do that as a follow up.

Reply:
will provide a separate PR

Reply:
hi, the test is in: https://github.com/leetcode-1533/presto-1/pull/1/files, I will create a PR for Trino once this PR merged.

Reply:
https://github.com/trinodb/trino/pull/15633